factor/grace/settings.php

```diff
@@ -37,6 +37,18 @@
     new lang_string('settings:forcesetup', 'factor_grace'),
     new lang_string('settings:forcesetup_help', 'factor_grace'), 0));
 
-$name = new lang_string('settings:graceperiod', 'factor_grace');
-$description = new lang_string('settings:graceperiod_help', 'factor_grace');
-$settings->add(new admin_setting_configduration('factor_grace/graceperiod', $name, $description, '604800'));
+$settings->add(new admin_setting_configduration('factor_grace/graceperiod',
+    new lang_string('settings:graceperiod', 'factor_grace'),
+    new lang_string('settings:graceperiod_help', 'factor_grace'), '604800'));
+
+// Get factors that gracemode cares about.
+$activefactors = \tool_mfa\plugininfo\factor::get_enabled_factors();
```

Comment:
will the setting screw up when we enable/disable factors and then save it? Maybe better to use all factors to be safe? and then check if it's enabled later (if required of course)...